sycl/test-e2e/Matrix/slm_utils.hpp

```diff
@@ -0,0 +1,163 @@
+          typename TOperand>
+inline void
+slm_read_write_OCL(TOperand *A, TOperand *B, local_accessor<TOperand, 2> tileA,
+                   local_accessor<TOperand, 2> tileB, sub_group sg,
+                   unsigned int k2, size_t m2, size_t n2, size_t SG_SIZE) {
+
+  size_t slmReadA = KCACHE2 / SG_SIZE;
+
+  for (int i = 0; i < (MCACHE2 / SGs); i++) {
+    __attribute__((opencl_global)) ushort *g_addrA;
+    g_addrA =
+        (__attribute__((opencl_global))
+         ushort *)(A +
+                   (m2 * MCACHE2 + (sg.get_group_id()) * (MCACHE2 / SGs) + i) *
+                       colsA +
+                   k2 * KCACHE2);
+    __attribute__((opencl_local)) ushort *l_addrA;
+    l_addrA =
+        (__attribute__((opencl_local)) ushort
+             *)(tileA.template get_multi_ptr<sycl::access::decorated::yes>() +
+                ((sg.get_group_id()) * (MCACHE2 / SGs) + i) * KCACHE2)
+            .get();
+    if (slmReadA == 2) {
+      __builtin_IB_simd_block_write_2_local_h(
+          l_addrA, __builtin_IB_simd_block_read_2_global_h(g_addrA));
+    } else if (slmReadA == 4) {
+      __builtin_IB_simd_block_write_4_local_h(
+          l_addrA, __builtin_IB_simd_block_read_4_global_h(g_addrA));
+    } else
+      assert(slmReadA == 2 || slmReadA == 4);
+  }
+  // how much each SG will load to SLM --> has to be contiguous
+  // NCACHE2*KCACHE2/(SGs*SG_SIZE) = 16
+  size_t slmRead = NCACHE2 * KCACHE2 / (SGs * SG_SIZE);
+  size_t sgsPerRow = (NCACHE2 * vnniFactor) / (slmRead * SG_SIZE);
+  __attribute__((opencl_global)) ushort *g_addr;
+  g_addr = (__attribute__((opencl_global))
+            ushort *)(B +
+                      (k2 * (KCACHE2 / vnniFactor) +
+                       (uint)(sg.get_group_id() / sgsPerRow)) *
+                          (colsB * vnniFactor) +
+                      n2 * NCACHE2 * vnniFactor +
+                      (sg.get_group_id() % sgsPerRow) * (slmRead * SG_SIZE));
+  __attribute__((opencl_local)) ushort *l_addr;
+  l_addr =
+      (__attribute__((opencl_local))
+       ushort *)(tileB.template get_multi_ptr<sycl::access::decorated::yes>() +
+                 ((uint)(sg.get_group_id() / sgsPerRow)) * NCACHE2 *
+                     vnniFactor +
+                 (sg.get_group_id() % sgsPerRow) * (slmRead * SG_SIZE))
+          .get();
+  if (slmRead == 16) {
+    __builtin_IB_simd_block_write_16_local_h(
+        l_addr, __builtin_IB_simd_block_read_16_global_h(g_addr));
+  } else if (slmRead == 8) {
+    __builtin_IB_simd_block_write_8_local_h(
+        l_addr, __builtin_IB_simd_block_read_8_global_h(g_addr));
+  } else
+    assert(slmRead == 8 || slmRead == 16);
+}
+
+#endif
+template <unsigned int rowsA, unsigned int colsA, unsigned int rowsB,
+          unsigned int colsB, unsigned int MCACHE2, unsigned int NCACHE2,
+          unsigned int KCACHE2, unsigned int vnniFactor, size_t SGs,
+          typename TOperand, access::address_space Space>
```

Comment:
```suggestion
          typename TOperand, access::address_space space>
```

Reply:
Variables start with an upper case letter

Reply:
I saw both ways in intel/llvm, but I believe it should be consistent at least across single component. If function parameters should start with an upper case, all of them should follow it. I see "colsA", "colsB", "vnniFactor". It should be corrected then.